chore(docs): ownership tracking concept and how-to (#2803)
open-component-model/open-component-model: website/content/docs/concepts/ownership.md

---
title: "Ownership"
description: "How OCM traces a resource back to the component version that owns it, without modifying the resource."
weight: 9
toc: true
hasMermaid: true
---

Ownership is about tracing a resource back to its owning component version. Find a resource in a registry and it
looks like any other image: nothing on it points back to the component version it belongs to. The component version

> lists its resources, but the resources say nothing about the component version.

Reviewer: `say` seems colloquial, maybe:

```suggestion
lists its resources, but the resources carry no information about the component version.
```

OCM's **ownership tracking** adds that missing reverse link. Given a resource, it lets you find the component version
that owns it.

## Always the Same Record

An ownership record carries no timestamp or other varying data, so its content is fully determined by the resource

> and the owner. The same resource owned by the same component version always produces the **same** record. Re-running

Reviewer: ```suggestion
and the owner. The same resource owned by the same component version always produces the **same** ownership record. Re-running
```

`ocm add component-version` lands on that same record; the registry sees a duplicate and keeps just one. A resource

> ends up with exactly **one** ownership record, however many times you create it.

Reviewer: Rest of the paragraph is third person
```suggestion
ends up with exactly **one** ownership record, however many times it is created.
```

## Ownership Across Transfer

> When you transfer a component version to another registry, OCM brings the ownership records for **by-value**

Reviewer 2: no mention that by-reference resources also get an ownership referrer - compared to the how-to, I feel we should mention it here as well, right?

Author: @matthiasbruns I changed it so that we have two resources, one not owned and the other owned, both byReference. The alter will get the ownership attachment.

> When you transfer a component version to another registry, OCM brings the ownership records for **by-value**

Reviewer: ```suggestion
When you transfer a component version to another registry, OCM brings the ownership records of **by-value**
```

> resources along automatically. The record travels together with the resource it points at, so it arrives whether or not you also copy the resources (`--copy-resources`).

Reviewer 2: is the `--copy-resources` correct?

Author: @matthiasbruns updated
